cloudinit/config/schema.py

```diff
@@ -1170,34 +1203,24 @@ def get_schema_dir() -> str:
     return os.path.join(os.path.dirname(os.path.abspath(__file__)), "schemas")
 
 
-def get_schema() -> dict:
-    """Return jsonschema coalesced from all cc_* cloud-config modules."""
-    # Note versions.schema.json is publicly consumed by schemastore.org.
-    # If we change the location of versions.schema.json in github, we need
-    # to provide an updated PR to
-    # https://github.com/SchemaStore/schemastore.
+def get_schema(schema_type: Optional[str] = "cloud-config") -> dict:
```

Comment:
Is `schema_type` actually Optional? We're using it as an index into `SCHEMA_FILES_BY_TYPE`, so if it is `None` we'll get a KeyError.